### PR TITLE
Refactor fix_data.yaml

### LINKED ISSUE #114168: Split up fix_data.yaml file
The fix data file has grown very large an d is difficult to parse.
Following up on https://github.com/dart-lang/sdk/issues/47941, it can now be split into multiple files.
We should split it up so it is cleaner and easier to work with.

### PR DESCRIPTION
**Pull request by Piinks**
We can now split this file up for easier everything. 🎉 
Fixes https://github.com/flutter/flutter/issues/114168

ToDo:
- ~rearrange tests equally~
- update wiki

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Piinks** [comment]
@asashour I've tried to split up our fix_data.yaml into multiple files but it is not working. :( 
I followed https://dart-review.googlesource.com/c/sdk/+/260581 but none of the fixes are being applied. Is a fix_data.yaml file still required?

**Comment by asashour** [comment]
`lib/fix_data.yaml` file is optional, and `lib/fix_data` folder is optional (which may recursively contain any other .yaml files to be processed.

**Comment by Piinks** [comment]
Oh thank goodness, I was hoping I had made a silly mistake. Thank you!

**Comment by Piinks** [comment]
This is blocked on https://github.com/dart-lang/sdk/issues/50358

File: packages/flutter/test_fixes/material/app_bar_theme.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  appBarTheme = appBarTheme.copyWith(brightness: Brightness.light);
+  appBarTheme = appBarTheme.copyWith(brightness: Brightness.dark);
+  appBarTheme.brightness;
+
+  AppBarTheme appBarTheme = AppBarTheme();
+  appBarTheme = AppBarTheme(textTheme: myTextTheme);
+  appBarTheme = AppBarTheme(textTheme: myTextTheme);
+  appBarTheme = appBarTheme.copyWith(textTheme: myTextTheme);
+  appBarTheme = appBarTheme.copyWith(textTheme: myTextTheme);
+
```
Comment (Swiftaxe) [inline packages/flutter/test_fixes/material/app_bar_theme.dart:23]: The variable 'myTextTheme' seems undeclared in the current scope. 
The declaration has been moved to app_bar.dart:15.

**Review by Swiftaxe — DISMISSED** [review]
Well done! 😃 

I can find only one issue, located in app_bar_theme.dart. See line comment.

**Comment by Piinks** [comment]
Thank you so much for reviewing this! You have a great eye!

**Review by pdblasi-google — APPROVED** [review]
LGTM! Such organization, wow!

**Comment by Swiftaxe** [comment]
> Thank you so much for reviewing this! You have a great eye!

Thanks! Pretty big PR. It was not easy to spot! 😄

Great teamwork! 👌😃

**Pull request merged**